etna/datasets/internal_datasets.py

```diff
@@ -132,13 +158,13 @@ def load_dataset(
     if not _check_dataset_local(dataset_path) or rebuild_dataset:
         get_dataset_function(dataset_dir)
     if len(parts_) == 1:
-        data = pd.read_csv(
-            dataset_dir / f"{name}_{parts_[0]}.csv.gz",
-            compression="gzip",
-            header=[0, 1],
-            index_col=[0],
-            parse_dates=[0],
-        )
+        data, dataset_hash = read_dataset(dataset_path=dataset_dir / f"{name}_{parts_[0]}.csv.gz")
+        if dataset_hash != datasets_dict[name]["hash"][parts_[0]]:
+            warnings.warn(
```

Comment:
As I can see there are possible reasons for hash difference:
- User saved the dataset, updated the library and there is differerence between saved version and library expected version. For example, because we added some additional preprocessing before saving. In that cases the dataset should be reloaded to fix the issue.
- User saved the dataset which is newer than installed library has fixed. In that case the library could be updated to fix the issue (if there is such update).

I think current explanation doesn't give enough info.

Reply:
Suggestion:

Local hash and expected hash are different for {name} record part {part}. 
The first possible reason is that the local copy of the dataset is out of date. In this case you can try setting `rebuild_dataset=True` to rebuild the dataset.
The second possible reason is that the local copy of the dataset reflects a more recent version of the data than your version of the library. In this case you can try updating the library version.